### ISSUE TITLE
table and heading issue

### ISSUE THREAD

**Issue by starsareintherose** [blackhole89/notekit#111]
![2021-12-16 03-59-49 的屏幕截图](https://user-images.githubusercontent.com/37733075/146305929-aceaab39-a88c-4ea8-a307-741fb3fe176c.png)
`--` can not be recognized as headings and the table can not be shown. sidebar is transparent.

**Comment by starsareintherose** [comment]
can I adjust css under data/  to redefine the sidebar color?

**Comment by sp1ritCS** [comment]
try building commit 18b708da752c17deeef5129166b922cbd43fa8c6. The way the sidebar color is handled got changed recently and it doesn't work that well for a lot of themes.

**Comment by sp1ritCS** [comment]
tables are not supported rn (see #47)

and I think checking for -- as alternative to # for titles is quite hard using gsv.

**Comment by starsareintherose** [comment]
Thanks, it works. and I found the space to make some thing code does not work, such as this


    #include <stdio.h>
    enum colors {red,yellow,green}; 
    void f (enum color c);
    int main (void)
    {
    	enum color t =read;
    	scanf("%d",&t);
    	f(t);
    	return 0;	
    }
    void f(enum color c)
    {
    	printf("%d\n",c);
    }


the original text is, no  ` ``` `, just space
```

    #include <stdio.h>
    enum colors {red,yellow,green}; 
    void f (enum color c);
    int main (void)
    {
    	enum color t =read;
    	scanf("%d",&t);
    	f(t);
    	return 0;	
    }
    void f(enum color c)
    {
    	printf("%d\n",c);
    }

```

**Comment by blackhole89** [comment]
* As sp1ritCS said above, we unfortunately can't support the syntax where titles are underlined with ---/=== easily due to parser limitations. Since it's redundant and less flexible than leading #s, I think this should not be too much of a problem...?
* Table support would be nice, but is a pretty hard problem (almost "HTML-complete")
* Four-space-indented code spans are another Markdown feature that is pretty hard to parse due to its non-local properties. (If a line is indented by four spaces, is it part of a source code span or is it because it's a continuation of some other indented environment, such as a bulleted list?) Unlike the underline headings, I believe that this would in fact be possible to support with limitations (by parsing all other indentation-based block syntax correctly, and then taking indented stretches to be code when they have not been recognised as anything else), but the implementation would be a bit nasty, and I wasn't sure if it's worth it since four-indent code spans seem to be pretty limited (can't specify the language to highlight in, etc.).
* If you use source code listings a lot, I recommend ticking "enable syntax highlighting" in the Preferences page. This results in somewhat higher startup times, but lets you highlight any language supported by gedit in three-backtick code spans like ```cpp\n...\n```.
* By the way, did you build the program from source or download some (older?) prebuilt package? I notice that your Chinese text still renders as "sans-serif"/"gothic"/黑体-style font, but in a recent patch I changed the default css to choose a "serif"/宋体 one to match the Latin text. (`data/stylesheet.css` should say `font: 16px Charter, Bitstream Charter, Noto Serif, Noto Serif CJK JP, Noto Naskh Arabic, serif;`. Of course, you are free to tweak this.) If you are on an older build, I recommend updating.

You can tweak the schema which is used to parse Markdown by editing the spec which in most packages defaults to being loaded in `/usr/share/notekit/sourceview/markdown.lang` (or making a copy of that subtree in /usr/local/share/notekit), or could be in the `sourceview/` subfolder for a portable installation such as the tar.gz build. It follows the [GtkSourceView language spec format](http://getfr.org/pub/dragonfly-release/usr-local-share/doc/gtksourceview-3.0/lang-reference.html), though some special conventions (mostly context classes and special chars) that notekit uses to interface with Gsv need to be respected. If you can come up with a good way of improving some of the parsing limitations, I would be more than happy to accept a pull request.

**Comment by blackhole89** [comment]
Oh yeah, by the way, could you point me at a download location for theme that you are using? If you are/were using the most recent version, am not sure I understand the rendering issues that you are having with the sidebar. You should of course feel free to adjust the css to make it render correctly, but I'm slightly taken aback that the default worked so badly.

**Comment by starsareintherose** [comment]
try this https://github.com/vinceliuice/Graphite-gtk-theme
a wonderful theme, @blackhole89

**Comment by blackhole89** [comment]
I think there is an issue in how the Graphite theme interacts with alpha-channel windows with a transparent background (such as notekit). I'm not sure if the problem is on our side or the theme's. You can fix the problem by adding a line like
```css
.notekit .sidebar { background-color: rgba(12,34,56,1); }
```
to `stylesheet.css` (and potentially replacing the current line about `border-color`, which also seems to not work with Graphite), while replacing 12,34,56 with the appropriate background colour, or alternatively commenting out the line that says `override_background_color(Gdk::RGBA("rgba(0,0,0,0)"));` in `mainwindow.cpp` and rebuilding.

> if I can get a good command of C and gtk

Well, the point of the Gsv language specs is that you don't actually need to know either or touch the source code at all to edit them.

**Comment by sp1ritCS** [comment]
@starsareintherose I have some questions regarding your `notekit-git` package on AUR. Could you answer my comment I left on it, there?

**Comment by starsareintherose** [comment]
> I think there is an issue in how the Graphite theme interacts with alpha-channel windows with a transparent background (such as notekit). I'm not sure if the problem is on our side or the theme's. You can fix the problem by adding a line like
> 
> ```css
> .notekit .sidebar { background-color: rgba(12,34,56,1); }
> ```
> 
> to `stylesheet.css` (and potentially replacing the current line about `border-color`, which also seems to not work with Graphite), while replacing 12,34,56 with the appropriate background colour, or alternatively commenting out the line that says `override_background_color(Gdk::RGBA("rgba(0,0,0,0)"));` in `mainwindow.cpp` and rebuilding.
> 
> (edit: Either of these should also remove the need of building from an older commit.)
> 
> > if I can get a good command of C and gtk
> 
> Well, the point of the Gsv language specs is that you don't actually need to know either or touch the source code at all to edit them.

Thanks, it is solved. and if I have time to learn about this, I will contribute to it.
